src/global_styling/functions/logicals.ts

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { keysOf } from '../../components/common';
+
+export const logicalSide = {
+  left: 'inline-start',
+  right: 'inline-end',
+  top: 'block-start',
+  bottom: 'block-end',
+  horizontal: 'inline',
+  vertical: 'block',
+};
+
+export const LOGICAL_SIDES = keysOf(logicalSide);
+export type LogicalSides = typeof LOGICAL_SIDES[number];
+
+const logicalMargins = {
+  'margin-left': 'margin-inline-start',
+  'margin-right': 'margin-inline-end',
+  'margin-top': 'margin-block-start',
+  'margin-bottom': 'margin-block-end',
+  'margin-horizontal': 'margin-inline',
+  'margin-vertical': 'margin-block',
+};
+
+const logicalPaddings = {
+  'padding-left': 'padding-inline-start',
+  'padding-right': 'padding-inline-end',
+  'padding-top': 'padding-block-start',
+  'padding-bottom': 'padding-block-end',
+  'padding-horizontal': 'padding-inline',
+  'padding-vertical': 'padding-block',
+};
+
+const logicalPosition = {
+  top: 'inset-inline-start',
+  right: 'inset-block-start',
+  bottom: 'inset-inline-end',
+  left: 'inset-block-end',
+  horizontal: 'inset-block',
+  vertical: 'inset-inline',
+  inset: 'inset',
+};
+
+const logicalSize = {
+  height: 'block-size',
+  width: 'inline-size',
+  'max-height': 'max-block-size',
+  'max-width': 'max-inline-size',
+  'min-height': 'min-block-size',
+  'min-width': 'min-inline-size',
+};
+
+const logicalOverflow = {
+  'overflow-x': 'overflow-block',
+  'overflow-y': 'overflow-inline',
+};
+
+const logicalRadius = {
+  'border-top-left-radius': 'border-start-start-radius',
+  'border-top-right-radius': 'border-start-end-radius',
+  'border-bottom-left-radius': 'border-end-start-radius',
+  'border-bottom-right-radius': 'border-end-end-radius',
+};
+
+export const logicals = {
+  ...logicalMargins,
+  ...logicalPaddings,
+  ...logicalPosition,
+  ...logicalSize,
+  ...logicalOverflow,
+  ...logicalRadius,
+};
+
+export const LOGICAL_PROPERTIES = keysOf(logicals);
+export type LogicalProperties = typeof LOGICAL_PROPERTIES[number];
+
+/**
+ *
+ * @param property A string that is a valid CSS logical property
+ * @param value String to output as the property value
+ * @returns `string` Returns the logical CSS property version for the given `property: value` pair
+ */
+export const logicalCSS = (property: LogicalProperties, value?: any) => {
```

Comment:
I think we can be more specific here with the `value` and type it as `string`, based on the options I'm seeing on various docs - WDYT? Also, does it really make sense for `value` to be optional? When would we ever want to output CSS that's empty?

```suggestion
export const logicalCSS = (property: LogicalProperties, value: string) => {
```

Reply:
I did have it as `string` once, but then I ran into this conflicting TS issue:
<img width="850" alt="Screen Shot 2022-05-02 at 15 56 27 PM" src="https://user-images.githubusercontent.com/549577/166316516-85a281c6-07ef-455c-901d-2caad883b440.png">

Basically when we set a lot of keys to be `CSSProperties` based, users can also just use simple numbers. So we can't be certain whether `value` will be exactly a string. I felt it can be fairly tolerable and they'd have to understand CSS values. 

Otherwise we'll have to go in and change some of our global types.

---

I also had it optional in case someone setup of an array and mapped through values including a `none: undefined` option.

Reply:
Ahhh gotcha. No worries then if it's a downstream type headache.